test/e2e/framework.go

```diff
@@ -1343,7 +1343,21 @@ func (data *TestData) restartAntreaAgentPods(timeout time.Duration) error {
 		return fmt.Errorf("error when deleting antrea-agent Pods: %v", err)
 	}
 
-	return data.waitForAntreaDaemonSetPods(timeout)
+	if err := data.waitForAntreaDaemonSetPods(timeout); err != nil {
+		var builder strings.Builder
+		pods, errGet := data.clientset.CoreV1().Pods(antreaNamespace).List(context.TODO(), metav1.ListOptions{
+			LabelSelector: "app=antrea,component=antrea-agent",
+		})
+		if errGet != nil {
+			builder.WriteString(errGet.Error())
+		}
+		for _, pod := range pods.Items {
+			code, stdout, stderr, errCmd := provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl -n %s logs %s antrea-agent", antreaNamespace, pod.Name))
+			builder.WriteString(fmt.Sprintf("RunCommandOnNode, code: %d, stdout: %s, stderr: %s, error: %v", code, stdout, stderr, errCmd))
+		}
+		return fmt.Errorf("restartAntreaAgentPods error： %v, logs: %s", err, builder.String())
+	}
```

Comment:
this seems unrelated to the rest of the PR and it's unclear what the purpose is since there is no comment
if there is an issue with the e2e test framework, it should be addressed in a separate PR

Reply:
When the agent starting failed it just describe the agent pods, I think we need to print agent starting log and makes it easier to debug.

Reply:
Agree with @antoninbas. And we already collect the component logs in specific directory if the test fails. I'm not sure if it will collect logs when deploying fails. But even if it doesn't, it's easy to change its behavior to do it. Dumpping the whole log in the error may mess up the test output.

Reply:
If we don't have the same test environment. It`s hard to find the panic cause, because the describe pod logs in Jenkins CI output just say 'Back off....'.
```
          Normal   Created    116s (x4 over 2m53s)  kubelet            Created container antrea-agent
          Normal   Started    116s (x4 over 2m53s)  kubelet            Started container antrea-agent
          Warning  BackOff    112s (x7 over 2m48s)  kubelet            Back-off restarting failed container
```